include/stream.h

```diff
@@ -296,7 +296,7 @@ class StreamCursor {
 
     };
 
-    bool advance(size_t count);
+    bool advance(ssize_t count);
```

Comment:
bacause the value will be compared somewhere and the compile warns that you compare signed value with an unsigned one. e.g. hardware_concurrency() returns init which is a signed value.

Also you could make all size_t, but you should decide which one to use and stick to it